File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.js
function solution(nums) {
    count = new Array(3000).fill(true);
    count[0] = count[1] = false;

    for (i = 2; i < 3000; i++) {
        if (count[i]) {
            for (j = i * i; j < 3000; j += i) {
                count[j] = false;
            }
        }
    }

    countPrime = 0;

    for (i = 0; i < nums.length - 2; i++) {
        for (j = i + 1; j < nums.length - 1; j++) {
            for (k = j + 1; k < nums.length; k++) {
                sum = nums[i] + nums[j] + nums[k];
                if (count[sum]) {
                    countPrime++;
                }
            }
        }
    }

    return countPrime;
}

File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.py
from itertools import *

def solution(nums):
    
    #소수 목록 구하기
    count = [True] * 3000
    count[0] = count[1] = False
    
    for i in range(2, 3000):
        if count[i]:
            for j in range(i*i, 3000, i):
                count[j] = False
    
    #세 수를 합해서 소수인지 확인하기
    numList = list(combinations(nums, 3))
    array = 0
    
    for i in range(0, len(numList)):
        if(count[sum(numList[i])]):
            array += 1
    
    return array